### PR TITLE
Make coordinate dimensions and derived quantities properties with set…

### PR DESCRIPTION
**Pull request by moeyensj**
…ters, getters, and deleters. 

Turns out these are fairly useful to define especially for subclasses.

### CODE REVIEW

File: adam_core/coordinates/cartesian.py [hunk 1]
```diff
@@ -82,126 +82,362 @@ def __init__(
...
+
+    @property
     def y(self):
         """
         Y-coordinate
         """
         return self._values[:, 1]
 
+    @y.setter
+    def y(self, value):
```
Comment (akoumjian) [inline adam_core/coordinates/cartesian.py:109]: When we set the value do we need to remove a possible mask?

File: adam_core/coordinates/cartesian.py [hunk 2]
```diff
@@ -82,126 +82,362 @@ def __init__(
...
+        self._values[:, 1] = value
+
+    @y.deleter
+    def y(self):
```
[the remark this answers is not in the document]
Comment (moeyensj) [reply adam_core/coordinates/cartesian.py:113]: Underlying arrays for things like observations are masked arrays to support instances of missing data. For example, most optical astronomical observations will only be RA, Dec (lon, lat), so (rho, vrho, vlon, vlat) will be masked out. Deleting a coordinate dimension seems like it should behave like masking them out after removing the actual values.

File: adam_core/coordinates/cartesian.py (hunk 1 above)
Comment (moeyensj) [reply adam_core/coordinates/cartesian.py:109]: It really should remove the mask to make the behavior consistent with the deleter. I'll fix it!

File: adam_core/coordinates/keplerian.py [hunk 3]
```diff
@@ -78,114 +78,312 @@ def __init__(
...
+
+    @property
     def mu(self):
         """
         Gravitational parameter
         """
         return self._mu
 
+    @mu.setter
+    def mu(self, value):
```
[the remark this answers is not in the document]
Comment (moeyensj) [reply adam_core/coordinates/keplerian.py:373]: Yes!

**Pull request merged**